File: cases/taint/kotlin/alias-propagation-positive/AliasPropagationPositive.kt
package dataflowbench

object AliasPropagationPositive {
    class Holder {
        var value: Int = 0
    }

    fun dfb_source(): Int { // DFB-SOURCE: alias-propagation-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: alias-propagation-sink

    fun run() {
        val original = Holder()
        val alias = original // DFB-WITNESS: alias-propagation-alias
        val distinct = Holder()
        original.value = dfb_source() // DFB-WITNESS: alias-propagation-store
        dfb_sink(alias.value)
    }
}

File: cases/taint/csharp/alias-propagation-positive/AliasPropagationPositive.cs
namespace DataFlowBench;

static class AliasPropagationPositive
{
    sealed class Holder
    {
        public int Value;
    }

    static int dfb_source() // DFB-SOURCE: alias-propagation-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: alias-propagation-sink

    static void Run()
    {
        Holder original = new Holder();
        Holder alias = original; // DFB-WITNESS: alias-propagation-alias
        Holder distinct = new Holder();
        original.Value = dfb_source(); // DFB-WITNESS: alias-propagation-store
        dfb_sink(alias.Value);
    }
}
